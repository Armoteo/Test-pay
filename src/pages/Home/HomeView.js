import { useSelector, useDispatch } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { fetchJokeAsync } from '../../store/jokes/slice';
import Button from '../../components/Button/Button';
import Loader from '../../components/Loader/Loader';
import styles from './sass/Home.module.scss';

const HomeView = () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const { isLoading, joke, error } = useSelector(({ jokes }) => jokes);
  return (
    <div>
      <p className={styles.appIntro}>
        {t('homePage')}
      </p>
      {isLoading ? (
        <Loader />
      ) : (
        <div>
          <p>{error || joke}</p>
        </div>
      )}
      <Button
        title={t('newJoke')}
        buttonClass={styles.updateJoke}
        handleClick={() => dispatch(fetchJokeAsync())}
      />
    </div>
  );
};

export default HomeView;
