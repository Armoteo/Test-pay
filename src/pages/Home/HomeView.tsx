import { useTranslation } from 'next-i18next';
import Layout from '../../layout/Layout';
import styles from './sass/Home.module.scss';

function HomeView() {
  const { t } = useTranslation('home');

  return (
    <Layout>
      <div className={styles.page}>
        {t('hello')}
      </div>
    </Layout>
  );
}

export default HomeView;
