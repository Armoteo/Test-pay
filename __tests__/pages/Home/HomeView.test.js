import userEvent from '@testing-library/user-event';
import { act } from '@testing-library/react';
import fetchMock from 'jest-fetch-mock';
import { render } from '../../settings/test-utils';
import HomeView from '../../../src/pages/Home/HomeView';

describe('<HomeView />', () => {
  let wrapper;
  const mockGetJoke = jest.fn();
  const mockedJoke = 'mock data';

  beforeEach(async () => {
    fetchMock.mockResponseOnce(JSON.stringify({ value: mockedJoke }));
    wrapper = await act(() => (
      render(
        <HomeView />
      )
    ));
  });

  test('should render HomeView component', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render Button component', () => {
    const { getByTestId } = wrapper;
    const Button = getByTestId('button');
    expect(Button).toBeInTheDocument();
  });

  test('on Button click should call mockGetJoke function', async () => {
    const { getByTestId } = wrapper;
    const Button = getByTestId('button');
    wrapper.store.subscribe(mockGetJoke);
    await act(async () => {
      await userEvent.click(Button);
    });
    expect(mockGetJoke).toHaveBeenCalled();
    expect(wrapper.getByText(mockedJoke)).toBeInTheDocument();
  });
});
