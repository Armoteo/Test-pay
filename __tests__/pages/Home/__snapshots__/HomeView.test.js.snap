// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeView /> should render HomeView component 1`] = `
<DocumentFragment>
  <div>
    <p
      class="appIntro"
    >
      Home page
    </p>
    <div>
      <p />
    </div>
    <button
      class="updateJoke"
      data-testid="button"
      type="button"
    >
      New joke
    </button>
  </div>
</DocumentFragment>
`;
